packages/admin-panel/src/widgets/JsonEditor.js

```diff
@@ -4,40 +4,51 @@
  */
 
 import React from 'react';
-import { TextField } from '@tupaia/ui-components';
 import PropTypes from 'prop-types';
+import { JsonEditor as Editor } from 'jsoneditor-react';
+import 'jsoneditor-react/es/editor.min.css';
+import Typography from '@material-ui/core/Typography';
+import styled from 'styled-components';
+
+const Container = styled.div`
+  margin-bottom: 20px;
+`;
+
+const Label = styled(Typography)`
+  color: ${props => props.theme.palette.text.secondary};
+  font-size: 0.9rem;
+  line-height: 1.1rem;
+`;
+
+const HelperText = styled(Typography)`
+  color: ${props => props.theme.palette.text.secondary};
+  font-size: 0.75rem;
+  margin-top: 3px;
+  line-height: 1.66;
+`;
 
 export const JsonEditor = ({ inputKey, label, secondaryLabel, value, onChange }) => {
-  let editorValue;
-
-  switch (typeof value) {
-    case 'object':
-      editorValue = JSON.stringify(value, null, 2);
-      break;
-    case 'string':
-      try {
-        const object = JSON.parse(value);
-        editorValue = JSON.stringify(object, null, 2);
-      } catch (e) {
-        editorValue = value;
-      }
-      break;
-    default:
-      editorValue = value.toString();
+  if (!value) {
+    return null;
```

Comment:
Is there a case where we want to allow creating a new json object, where previously the column was null?

Reply:
No, it is in case the component gets mounted with no data in which case the json editor will throw an error. Another way to handle this would be to set value to an empty object which now I think about it might be better in case there is a null value from the db.

Reply:
Yeah I reckon that's best :+1:

Reply:
hmmm so the json-editor handles it's state internally so the value prop is actually just the default value. When the component first renders, the data hasn't come back form the server yet and is null, then when the server responds the editor will not update with the new data. The editor needs to be updated by user input. The null check fixes this issue but I guess there could be cases where this is a problem.

Reply:
Looking through the database, all the empty values for json fields are `{}` which works fine. I'm not sure if that is a default somewhere or not.

Reply:
We use that as a default in most cases - sounds like maybe all, so that's good. There are some json strings stored in the db as text fields, and I'm not sure but we might also use the json editor to handle them, and I'm not sure whether they all have the same default. Can just keep it as is for now and log the bug if it comes up haha